php/base/Exchange.php

```diff
@@ -1346,6 +1346,14 @@ function ($curl, $header) use (&$response_headers, &$http_status_text) {
         $curl_error = curl_error($this->curl);
         $http_status_code = curl_getinfo($this->curl, CURLINFO_HTTP_CODE);
 
+        // Write cookies here but do not close the handle
+        $cookie_file = static::safe_value($this->curl_options, CURLOPT_COOKIEJAR);
+        if ($cookie_file) {
+            $cookies = implode("\n", curl_getinfo($this->curl, CURLINFO_COOKIELIST)) . "\n";
+            $file = fopen($cookie_file, 'w');
```

Comment:
We're not really supposed to do file io or disk io inside the lib... Instead, I would suggest to merge the options that are necessary to handle the cookie file by means of curl – with the other options we're setting, and without resetting the handle (only enable this if the corresponding controlling properties are set on the exchange instance). And let the user handle file io on the user side. Lmk what you think.

Reply:
curl does file io to the cookie_jar when curl_close is called if `CURLOPT_COOKIEJAR` is set. I just replicated their design.

Reply:
`CURLOPT_COOKIEFILE` is the file it reads the cookies from and `CURLOPT_COOKIEJAR` is where it stores the cookies too upon the closing of the handle.

Reply:
> curl does file io to the cookie_jar when curl_close is called if CURLOPT_COOKIEJAR is set. I just replicated their design.

Yes, but why won't we just leave it for CURL to do that, instead of actually adding fopen to the base code?

Reply:
because we don't want to close the handle

Reply:
there is no `curl_save_cookies` function just `curl_close` afaik

Reply:
> because we don't want to close the handle

Do we have to?

What about these? ↓

- https://www.php.net/manual/en/function.curl-setopt.php#114475
- https://www.php.net/manual/en/function.curl-setopt.php#118967
- https://www.php.net/manual/en/function.curl-setopt.php#91510

Reply:
We don't close the handle in the current version.

Reply:
`curl_close($this-curl)` would throw an error

Reply:
What kind of error?